src/main/webapp/IFrame.js

```diff
@@ -12,143 +12,189 @@
+const SYNC_WINDOW = 5; // max time diff between client and Firestore
+
+const firebaseConfig = {
+  apiKey: API_KEY, // eslint-disable-line no-undef
+  authDomain: 'lounge-95f01.firebaseapp.com',
+  databaseURL: 'https://lounge-95f01.firebaseio.com',
+  projectId: 'youtube-lounge',
+  storageBucket: 'youtube-lounge.appspot.com',
+  messagingSenderId: '681171972170',
+  appId: '1:681171972170:web:4c6526b8eb788af9d876b3',
+  measurementId: 'G-JSDHBSMHS3',
+};
+firebase.initializeApp(firebaseConfig); // eslint-disable-line no-undef
+const firestore = firebase.firestore(); // eslint-disable-line no-undef
+
+// Hard coded for now, but eventually will update to point
+// to current video in queue
+const docRef = firestore.doc('CurrentVideo/PlaybackData');
+
+let player; // var representing iframe ytplayer
 function onYouTubeIframeAPIReady() { // eslint-disable-line no-unused-vars
   player = new YT.Player('ytplayer', { // eslint-disable-line no-undef
     events: {
       'onStateChange': onPlayerStateChange,
       'onPlaybackRateChange': onPlayerPlaybackRateChange,
     },
   });
+  // I'm still working on writing this as a promise, but it doesn't
+  // work unless these next two parts are delayed
+  setTimeout(function() {
+    sendInfo('start');
+    getRealtimeUpdates();
+  }, 1000);
 }
 
-function sendInfo(playing) {
-  const xhttp = new XMLHttpRequest();
-  xhttp.onreadystatechange = function() {
-    if (this.readyState == 4 && this.status == 200) {
-      console.log('Request sent');
-    }
-  };
-  xhttp.open('POST', '/Playback-Test', true);
-  xhttp.setRequestHeader('Content-type', 'application/x-www-form-urlencoded');
-  const time = player.getCurrentTime();
-  const speed = player.getPlaybackRate();
-  xhttp.send('timestamp=' + time + '&videoSpeed=' + speed + '&isPlaying='+
-    playing);
+function sendInfo(goal) { // send info to Firestore
+  docRef.set({
+    isPlaying: isVideoPlaying(),
+    timestamp: player.getCurrentTime(),
+    videoSpeed: player.getPlaybackRate(),
+  }).then(function() {
+    console.log(goal + ' request sent');
+  }).catch(function(error) {
+    console.log(goal + ' caused an error: ', error);
+  });
 }
 
-let timeout;
-function onPlayerStateChange() {
-  console.log(player.getPlayerState());
-  if (!videoUpdating) {
-    switch (player.getPlayerState()) {
-      case 1: // Playing
-        clearTimeout(timeout);
-        sendInfo(true);
-        break;
-      case 2: // paused
-        timeout = setTimeout(sendInfo, 100, false);
-        break;
-      case 3: // Buffering
-        clearTimeout(timeout);
-        break;
-      case -1: // Just before video starts
-        beginFetchingLoop();
-        break;
-      case 0: // Ended
-        endFetchingLoop();
-    }
-  }
-}
+let catchUp = false; // Does this vid need to catch up to Firestore?
 
-// Get the time of the current video
-function getTime() { // eslint-disable-line no-unused-vars
-  console.log(player.getCurrentTime());
-}
+/*
```

Comment:
Could we maybe shorten this comment? Seems like we're giving up some implementation details that may not be necessary

Reply:
Sure thing!